packages/venia-ui/lib/components/CategoryTree/categoryTree.js

```diff
@@ -33,7 +33,7 @@ const Tree = props => {
         const [id, { category, isLeaf }] = childCategory;
 
         return isLeaf ? (
-            <Leaf key={id} category={category} onNavigate={onNavigate} />
+            <Leaf category={category} onNavigate={onNavigate} />
```

Comment:
```suggestion
            <Leaf key={id} category={category} onNavigate={onNavigate} />
```

This key is needed, but does seem to introduce a duplicate key or fetching problem where the parent leaf is rendered while the tree is fetching.

![](https://gyazo.com/52a947f1f784f888d262d9c710d96b16.gif)

Reply:
Thanks @tjwiebell  for your feedback. I have just updated key in Leaf.